sdl_android/src/main/java/com/smartdevicelink/api/audio/BaseAudioDecoder.java

```diff
@@ -0,0 +1,226 @@
+package com.smartdevicelink.api.audio;
+
+import android.content.Context;
+import android.media.AudioFormat;
+import android.media.MediaCodec;
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+import com.smartdevicelink.api.audio.AudioStreamManager.SampleType;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+@SuppressWarnings("WeakerAccess")
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public abstract class BaseAudioDecoder {
+    private static final String TAG = AudioDecoder.class.getSimpleName();
+
+    protected int targetSampleRate;
+    protected @SampleType int targetSampleType;
+
+    private int outputChannelCount;
+    private int outputSampleRate;
+    private @SampleType int outputSampleType;
+
+    private double lastOutputSample = 0;
+
+    private double lastOutputPresentationTimeUs = 0;
+    private double lastTargetPresentationTimeUs = 0;
+
+    protected MediaExtractor extractor;
+    protected MediaCodec decoder;
+
+    protected Uri audioSource;
+    protected Context context;

```

Comment:
This should be `private final WeakReference<Context>`. Even though this class should only have a short runtime, it's important to note its use in a queue which means it could hang around for an extended time and therefore shouldn't hold a direct reference to a context.